lambda/onomondo/getAllUsedSimsOnomondo.ts

```diff
@@ -1,42 +1,70 @@
 import { fetchAndValidate } from '../fetchAndValidate.js'
 import { Type, type Static } from '@sinclair/typebox'
 
-export const getAllUsedSimsOnomondo = async (
-	apiKey: string,
-): Promise<Array<string>> => {
-	const iccids: Array<string> = []
-	const filter = `time:${new Date().toISOString().slice(0, 10)}`
+export type simInfoTS = {
+	usedBytes: number
+	billId: string
+}
+
+export const getSimUsageHistoryOnomondo = async ({
+	apiKey,
+	iccid,
+	date,
+}: {
+	apiKey: string
+	iccid?: string
+	date?: Date
+}): Promise<Record<string, Record<string, simInfoTS>>> => {
+	const res: Record<string, Record<string, simInfoTS>> = {}
+	const filter = `time:${(date ?? new Date()).toISOString().slice(0, 10)}`
 	const endpoint = 'https://api.onomondo.com/'
 	let hasMore = true
 	const searchParam: { next_page?: string; filter: string } = { filter }
+	let string = ''
+	if (iccid !== undefined) {
+		string = `/${iccid}`
+	}
 	while (hasMore) {
```

Comment:
Generally we do not use while loops for pagination. Refactor this into a recursive implementation, because this makes it harder to accidentally created an endless loop.